File: backend/src/com/origin/Launcher.java
package com.origin;

import com.origin.net.GameServer;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.net.InetSocketAddress;
import java.util.Locale;

public class Launcher
{
	private static final Logger _log = LoggerFactory.getLogger(Launcher.class.getName());

	public static void main(String... args)
	{
		Locale.setDefault(Locale.ROOT);

		ServerConfig.loadConfig();
		Database.start();

		_log.debug("start game server...");

		GameServer server = new GameServer(new InetSocketAddress("0.0.0.0", 7070), Runtime.getRuntime().availableProcessors());
		server.start();
	}
}

File: backend/src/com/origin/Launcher.kt
package com.origin

import com.origin.database.DatabaseFactory
import com.origin.net.GameServer
import io.ktor.util.*
import kotlinx.coroutines.ObsoleteCoroutinesApi
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import java.util.*


@ObsoleteCoroutinesApi
object Launcher {
    val logger: Logger = LoggerFactory.getLogger(Launcher::class.java)

    @KtorExperimentalAPI
    @JvmStatic
    fun main(args: Array<String>) {
        Locale.setDefault(Locale.ROOT)
        ServerConfig.load()
        FileWatcher.start()
        EventBus.init()
        DatabaseFactory.init()
        TimeController.start()
        GameServer.start()
    }
}